### ISSUE TITLE
Environment replacement yields parse errors

### ISSUE THREAD

**Issue by jbergstroem** [hadolint/hadolint#179]
Ran into parsing issues while setting environment variables (ref: https://docs.docker.com/engine/reference/builder/#environment-replacement)

- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

A valid pass

### Actual behavior

```console
/dev/stdin:7:57 unexpected "\n" expecting "\\ ", "\\" or "="
```

### Steps to reproduce the behavior

Try parsing a dockerfile with the following content:
```dockerfile
ARG BUNDLE_WITHOUT
ENV BUNDLE_WITHOUT=${bundle_without:-'development test'}
```

**Comment by lorenzo** [comment]
This is a case I chose to not implement, as could only find one (invalid) use case across all popular dockerfiles in GitHub. I'm happy to add support for it. though.

I just have one question about it: Why not do this instead?

```dockerfile
ARG BUNDLE_WITHOUT='development test'
ENV BUNDLE_WITHOUT=${bundle_without}
```

**Comment by jbergstroem** [comment]
@lorenzo would work just as well -- just wanted to share an error I found in a clients gitlab pipeline.

**Comment by lorenzo** [comment]
Thanks for reporting @jbergstroem. I just wanted to make sure you were not blocked by this issue
